src/Cell.cpp

```diff
@@ -250,6 +287,56 @@ std::map<int, surface_halfspace> Cell::getSurfaces() const {
 
 
 /**
+ * @brief Returns the std::map of Cell IDs and Cell pointers within any
+ *        nested Universes filling this Cell.
+ * @return std::map of Cell IDs and pointers
+ */
+std::map<int, Cell*> Cell::getAllCells() {
+
+  std::map<int, Cell*> cells;
+
+  if (_cell_type == FILL && _fill != NULL) {
+    std::map<int, Cell*> nested_cells;
+    Universe* univ_fill = static_cast<Universe*>(_fill);
+
+    if (univ_fill->getType() == SIMPLE)
+      nested_cells = univ_fill->getAllCells();
+    else
+      nested_cells = static_cast<Lattice*>(univ_fill)->getAllCells();
+
+    cells.insert(nested_cells.begin(), nested_cells.end());
+  }
+
+  return cells;
+}
```

Comment:
For what used to be a `CellBasic`, this should either return itself or throw an error.  Unless there's a reason to return an empty map?